modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2TreeIndexBase.java

```diff
@@ -90,17 +96,35 @@ protected H2TreeIndexBase(GridH2Table tbl, String name, IndexColumn[] cols, Inde
         return true;
     }
 
+    /**
+     * To avoid tree corruption, at least two items should fit into one page.
```

Comment:
Why does a single item lead a tree corruption?

Reply:
https://support.gridgain.com/a/tickets/20077

Such trees are unbalanced, and they can overflow the max height of 128 layers